api/base/exceptions.py

```diff
@@ -80,20 +88,87 @@ class Conflict(APIException):
     default_detail = ('Resource identifier does not match server endpoint.')
 
 
-class InvalidQueryStringError(JSONAPIException):
+class JSONAPIParameterException(JSONAPIException):
+    def __init__(self, detail=None, parameter=None):
+        source = {
+            'parameter': parameter
+        }
+        super(JSONAPIParameterException, self).__init__(detail=detail, source=source)
+
+
+class JSONAPIAttributeException(JSONAPIException):
+    def __init__(self, detail=None, attribute=None):
+        source = {
+            'pointer': '/data/attributes/{}'.format(attribute)
+        }
+        super(JSONAPIAttributeException, self).__init__(detail=detail, source=source)
+
+
+class InvalidQueryStringError(JSONAPIParameterException):
     """Raised when client passes an invalid value to a query string parameter."""
     default_detail = 'Query string contains an invalid value.'
     status_code = http.BAD_REQUEST
 
-    def __init__(self, detail=None, parameter=None):
-        super(InvalidQueryStringError, self).__init__(detail=detail, source={
-            'parameter': parameter
-        })
 
+class InvalidFilterOperator(JSONAPIParameterException):
+    """Raised when client passes an invalid operator to a query param filter."""
+    status_code = http.BAD_REQUEST
+
+    def __init__(self, detail=None, value=None, valid_operators=None):
```

Comment:
Is there any reason to not specify the default `valid_operators` (as an immutable tuple) in the kwarg instead of setting with `or` below?